src/App.jsx

```diff
@@ -14,33 +17,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortGoods = (goods, params) => {
+  const copyGoods = [...goods];
+  const { sortedBy, isReversed } = params;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortedBy !== '') {
+    copyGoods.sort((good1, good2) => {
+      switch (sortedBy) {
+        case 'byAlphabet':
+          return good1.localeCompare(good2);
+        case 'byLength':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReversed) {
+    copyGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return copyGoods;
+};
+
+export const App = () => {
+  const [sortedBy, setSortedBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [hasChanged, setHasChanged] = useState(false);
+
+  const handleReset = () => {
+    setSortedBy('');
+    setIsReversed(false);
+    setHasChanged(false);
+  };
+
+  const sortedGoods = sortGoods(goodsFromServer, {
+    sortedBy,
+    isReversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortedBy('byAlphabet');
+            setHasChanged(true);
+          }}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortedBy !== 'byAlphabet',
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => {
+            setSortedBy('byLength');
+            setHasChanged(true);
+          }}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortedBy !== 'byLength',
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => {
+            setIsReversed(!isReversed);
+            setHasChanged(sortedBy !== '' || !isReversed);
```

Comment:
The logic for setting `hasChanged` is not functioning as intended. The expression `sortedBy !== '' || !isReversed` will always evaluate to `true` if `sortedBy` is a non-empty string. This could lead to unexpected behavior. Consider revisiting this logic to ensure it accurately reflects changes in sorting or reversal.